Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -2370,8 +2370,12 @@ void RenderBlock::computeBlockPreferredLogicalWidths(LayoutUnit& minLogicalWidth
         //
         // Floats inside a block level box may not use their margins in their
         // intrinsic size contributions if margin-trim is set for that margin type
-        Length startMarginLength = shouldChildInlineMarginContributeToContainerIntrinsicSize(MarginTrimType::InlineStart, *dynamicDowncast<RenderElement>(child)) ? childStyle.marginStartUsing(&styleToUse) : Length { 0, LengthType::Fixed };
-        Length endMarginLength = shouldChildInlineMarginContributeToContainerIntrinsicSize(MarginTrimType::InlineEnd, *dynamicDowncast<RenderElement>(child)) ? childStyle.marginEndUsing(&styleToUse) : Length { 0, LengthType::Fixed };
+        Length startMarginLength;
+        Length endMarginLength;
+        if (auto renderElement = dynamicDowncast<RenderElement>(child); renderElement && shouldChildInlineMarginContributeToContainerIntrinsicSize(MarginTrimType::InlineStart, *renderElement))
```

Comment:
I think the preferred way of writing this is
if (auto* renderElement<...

Reply:
Changed